src/GraphQL/StringExtensions.cs

```diff
@@ -64,5 +64,47 @@ private static string FastChangeFirstLetter(char newFirstLetter, string s)
             s.AsSpan().Slice(1).CopyTo(buffer.Slice(1));
             return buffer.ToString();
         }
+
+        /// <summary>
+        /// Returns a constant case version of this string. For example, converts 'StringError' into 'STRING_ERROR'.
+        /// </summary>
+        public static string ToConstantCase(this string value)
```

Comment:
The algorithm has become much clearer. As I see it, it is not difficult to change it for using spans. Will you do it? You just need to create a stackalloc/heap buffer and copy symbols to it one-by-one each step.

Reply:
Of cource in this case there will be no need to concatenate `value` and use `.Substring`.

Reply:
Buffer size can be roughly calculated in advance as `value.Length + value.Length/2`.

Reply:
I'm putting my effort into allowing input graph types to have resolvers.

Feel free to edit this branch as you see fit.

Reply:
> Buffer size can be roughly calculated in advance as `value.Length + value.Length/2`.

I was thinking that it would probably be very fast for the CPU to scan for matching patterns and then know exactly how large to make the buffer.  Performance testing and exact implementation would matter.  But with stackalloc it should not be necessary to know the exact length.

Reply:
Keep in mind that there's a 40% improvement to be made by switching IsUpper to `(char >= 'A' && char <= 'Z')` and similar, but then it does not match on foreign characters.  I opted to keep the function generic for all uses.

Reply:
I really just wanted to remove regex, as even this "slow" code represents a 20x improvement when using a string with 5 words.  (It is optimal if the string has one word, and should be nearly optimal if the string has two words, depending on how much .NET optimizes substring and concatenation.)

Reply:
It actually works better than the regex.  The regex fails one of the tests, and the regex does not match foreign upper/lowercase characters.